phpBB/phpbb/search/backend/base.php

```diff
@@ -445,19 +444,17 @@ protected function forum_ids_with_indexing_enabled(): array
 	}
 
 	/**
-	 * Get posts between 2 ids
+	 * Get batch of posts after id
 	 *
-	 * @param int $initial_id
-	 * @param int $final_id
+	 * @param int $post_id
 	 * @return \Generator
 	 */
-	protected function get_posts_between(int $initial_id, int $final_id): \Generator
+	protected function get_posts_batch_after(int $post_id): \Generator
 	{
 		$sql = 'SELECT post_id, post_subject, post_text, poster_id, forum_id
 			FROM ' . POSTS_TABLE . '
-			WHERE post_id >= ' . $initial_id . '
-				AND post_id <= ' . $final_id;
-		$result = $this->db->sql_query($sql);
+			WHERE post_id > ' . $post_id;
```

Comment:
if you do it in this way, i think you have to specify the order in the query

Reply:
in that case query builder will be advisable, Right ?

Reply:
i dont understand. what i mean is that you should make sure that posts are ordered by post_id

Reply:
Yes ! I meant via query builder `sql_build_query(...)`

Reply:
Never mind, I'll use order_by directly.
commit will be made soon.

Reply:
If this is an optimization could you share some numbers on how is this faster? `post_id` should be the primary key and thus an index, therefore I don't really see what the benefit here is. Although I could be just plain wrong.

Reply:
> If this is an optimization could you share some numbers on how is this faster? `post_id` should be the primary key and thus an index, therefore I don't really see what the benefit here is. Although I could be just plain wrong.

primary key is it's self an index,
but it can not be used to search post text.

also as-per previous code, between 2 post_ids there may or may not be 100/100 posts,
sometimes or most of the times some posts might be deleted and hence no of post can be less than 100, eg. 80/100.

I optimized such that there will be always be 100/100 posts to index it in one `while(...)` loop.

Reply:
yeah, I get that part, but is that an optimization in terms of performance? Or do you mean optimizing the behaviour?

Reply:
I am not an expert, but as-per me, it should be both.

Reply:
Example (Approx, since it's system depended):

Step is no of refresh,
inside each step there are no of `while(...)` loops.

In Previous code:
Step 1: 80 | 60 | 100 = 240
Step 2: 40 | 90 | 70 = 200
Step 3: 50 | 70 | 90 = 210

PR code:
Step 1: 100 | 100 | 100 = 300
Step 2: 100 | 100 | 100 = 300
Step 3: 50 = 50

also on larger board, even no of Steps will reduce.

I hope you understand what I mean, this is the best explanation I can come with.